unison-src/new-runtime-transcripts/_base.md

```diff
@@ -13,215 +13,7 @@ transcripts which contain less boilerplate.
-
-stdout = IO.stdHandle StdOut
-printText : Text -> {io2.IO} Either Failure ()
-printText t = putBytes.impl stdout (toUtf8 t)
-
--- Run tests which might fail, might create temporary directores and Stream out
--- results, returns the Results and the result of the test
-evalTest: '{Stream Result, TempDirs, io2.IO, Exception} a ->{io2.IO, Exception}([Result], a)
-evalTest a = handle (handle !a with Stream.collect.handler) with !autoCleaned.handler
-
--- Run tests which might fail, might create temporary directores and Stream out
--- results, but ignore the produced value and only return the test Results
-runTest: '{Stream Result, Exception, TempDirs, Exception, io2.IO} a -> {io2.IO}[Result]
-runTest t = handle evalTest t with cases
-    { Exception.raise (Failure _ f _) -> _ } -> [ Fail ("Error running test: " ++ f) ]
-    { (a, _) } -> a
-
-expect : Text -> (a -> a -> Boolean) -> a -> a -> {Stream Result} ()
-expect msg compare expected actual = if compare expected actual then emit (Ok msg) else emit (Fail msg)
-
-expectU : Text -> a -> a -> {Stream Result} ()
-expectU msg expected actual = expect msg (==) expected actual
-
-startsWith: Text -> Text -> Boolean
-startsWith prefix text = (eq (Text.take (size prefix) text) prefix)
-
-contains : Text -> Text -> Boolean
-contains needle haystack = if (size haystack) == 0 then false else
-  if startsWith needle haystack then true else
-    contains needle (drop 1 haystack)
-
-isDirectory = compose reraise isDirectory.impl
-createTempDirectory = compose reraise createTempDirectory.impl
-removeDirectory = compose reraise removeDirectory.impl
-fileExists = compose reraise fileExists.impl
-renameDirectory = compose2 reraise renameDirectory.impl
-openFile = compose2 reraise openFile.impl
-isFileOpen = compose reraise isFileOpen.impl
-closeFile = compose reraise closeFile.impl
-isSeekable = compose reraise isSeekable.impl
-isFileEOF = compose reraise isFileEOF.impl
-Text.fromUtf8 = compose reraise fromUtf8.impl
-getBytes = compose2 reraise getBytes.impl
-handlePosition = compose reraise handlePosition.impl
-seekHandle = compose3 reraise seekHandle.impl
-putBytes = compose2 reraise putBytes.impl
-systemTime = compose reraise systemTime.impl
-decodeCert = compose reraise decodeCert.impl
-serverSocket = compose2 reraise serverSocket.impl
-listen = compose reraise listen.impl
-handshake = compose reraise handshake.impl 
-send = compose2 reraise send.impl 
-closeSocket = compose reraise closeSocket.impl
-clientSocket = compose2 reraise clientSocket.impl
-receive = compose reraise receive.impl
-terminate = compose reraise terminate.impl
-newServer = compose2 reraise newServer.impl
-socketAccept = compose reraise socketAccept.impl
-socketPort = compose reraise socketPort.impl
-newClient = compose2 reraise newClient.impl
-MVar.take = compose reraise take.impl
-MVar.put = compose2 reraise put.impl
-MVar.swap = compose2 reraise MVar.swap.impl
-```
-
-
-```ucm:hide
+.> load unison-src/new-runtime-transcripts/base.u
```

Comment:
Switched to loading `base.u` from a file, which is handy when doing development of `base.u`.

Reply:
We have a working version of `base.u`??  Also what is `base.u`.

Reply:
It's just common code used by multiple transcripts so they don't all have to start with a prelude of utility functions. @stew added it a little while back and I just pulled out the unison code into a separate .u file

Reply:
Oh, I see, thanks.